Snake/BusinessLayer/Engine.cs

```diff
@@ -6,6 +6,11 @@
 
 namespace BusinessLayer
 {
+    public enum direction
+    {
+        right, down, left, up
```

Comment:
If these are still mapping to int's like we discussed you might want to consider explicitly defining that, for example - 
{
  Right = 1,
  Down = 2,
  ...
}
I would recommend rather than using the int to use the enum across all the classes.